extensions/kafka-client/deployment/src/main/java/io/quarkus/kafka/client/deployment/KafkaProcessor.java

```diff
@@ -105,6 +105,10 @@ public void build(CombinedIndexBuildItem indexBuildItem, BuildProducer<Reflectiv
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, RangeAssignor.class.getName()));
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, RoundRobinAssignor.class.getName()));
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, StickyAssignor.class.getName()));
+
+        // classes needed to perform reflection on DirectByteBuffer - only really needed for Java 8
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, "java.nio.DirectByteBuffer"));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, false, "sun.misc.Cleaner"));
```

Comment:
Wondering if that would fail on Java 11... The CI will let us now.

Reply:
It won't. I tried all combinations :)